js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class tuxexchange extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'tuxexchange',
+            'name': 'Tux Exchange',
+            'countries': ['CN'],
+            'version': '',
+            'accounts': undefined,
+            'accountsById': undefined,
+            'hostname': 'tuxexchange.com/api',
+            'has': {
+                'CORS': true, // Right? https://tuxexchange.com/api?method=getticker
+                'fetchCurrencies': true,
+                // 'fetchTradingLimits': true, // TODO: not documented, not responsive, might leave it out
+                'fetchTradingFee': true,
+                // 'fetchFundingLimits': true, // TODO: not documented, not responsive, might leave it out
+                'fetchTicker': true, // ALGO: I'm filtering fetchTickers, there is no direct endpoint, but I think this endpoint is important enough to leave this
+                'fetchTickers': true,
+                'fetchTrades': true,
+                'fetchBalance': true,
+                'createOrder': true,
+                'cancelOrder': true,
+                'fetchOpenOrders': true,
+                'fetchMyTrades': true,
+                'fetchDepositAddress': true,
+                'fetchWithdrawals': true,
+                'fetchDeposits': true,
+                'fetchClosedOrders': false,
+                'fetchL2OrderBook': false,
+                'fetchOHLCV': false,
+                'fetchOrder': false,
+                'editOrder': false,
+                'fetchTransactions': false,
+                'fetchLedger': false,
+                'withdraw': false,
+            },
+            'urls': {
+                'logo': 'https://www.tuxexchange.com/images/kittytrade.png',
+                'api': 'https://www.tuxexchange.com/api',
+                'www': 'https://www.tuxexchange.com',
+                'doc': 'https://www.tuxexchange.com/docs',
+                'fees': 'https://www.tuxexchange.com/faq',
+            },
+            'api': {
+                // All methods are passed in as query params
+                'public': { 'get': [''] },
+                'private': { 'post': [''] },
+            },
+            'fees': {
+                'trading': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'maker': 0,
+                    'taker': 0.3,
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const tickerRes = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (tickerRes);
+        const coinRes = await this.publicGet ({ 'method': 'getcoins' });
+        const coinResponse = this.parseJson (coinRes);
```

Comment:
I noticed that no other exchange implementations explicitly call `parseJson` -- why is that?

Reply:
ah, never did take another look at this. they seem to be sending back leading whitespace in the response. I'll add a trim() function in each of the base classes and see if tests are still good on other exchanges